### ISSUE TITLE
v3.10 Upgrade Fails on First Master on Atomic Host

### ISSUE THREAD

**Issue by ceagan** [openshift/openshift-ansible#9730]
#### Description

When running the latest 3.10 branch playbooks as of commit 35ceaa97f6b278d9366ec5069c996493ee74fa9f, the upgrade fails at Play _Update master nodes_, Task _Wait for master API to come back online_ for the first master.

My investigation so far has revealed that the _/etc/systemd/system/origin-master-api.service_ still exists and refers to _/etc/sysconfig/origin-master-api_. However, that file does not exist, so the service fails to start up.

##### Version

```
[root@openshift-ansible-installer openshift-ansible]# ansible --version
ansible 2.4.6.0

[root@openshift-ansible-installer openshift-ansible]# git describe
openshift-ansible-3.10.35-1-7-g35ceaa9
```

##### Steps To Reproduce
1. Run `ansible-playbook ~/openshift-ansible/playbooks/byo/openshift-cluster/upgrades/v3_10/upgrade_control_plane.yml`

##### Expected Results
Successful startup of master api service

##### Observed Results
Failure to start master api service.

journalctl output
```
Starting Atomic OpenShift Master API...
origin-master-api.service holdoff time over, scheduling restart.
Failed to load environment files: No such file or directory
origin-master-api.service failed to run 'start-pre' task: No such file or directory
Failed to start Atomic OpenShift Master API.
Unit origin-master-api.service entered failed state.
origin-master-api.service failed.
```

##### Additional Information

```
[root@openshift-master-2 ~]# cat /etc/redhat-release
CentOS Linux release 7.5.1804 (Core)
```

**Comment by ceagan** [comment]
I can see that _openshift-ansible/roles/openshift_control_plane/tasks/upgrade.yml_ contains a task that intentionally deletes _/etc/systemd/system/origin-master-api.service_, but on my system, that file still exists after the scripts run. There is also a task in that same file that moves the _/etc/sysconfig/origin-master-api_ to _/etc/sysconfig/origin-master-api.backup_ and migrates the configuration to a new _master.env_ file.

I expect that somewhere in the scripts, the _/etc/systemd/system/origin-master-api.service_ is re-generated with new settings, but I can't seem to find that script. The file I'm seeing in that location still looks like the old one to me. Note that for historical reasons, our openshift-master-2 node on this cluster is the first master, even though it has a 2 in the name.

```
[root@openshift-master-2 ~]# cat /etc/systemd/system/origin-master-api.service
[Unit]
Description=Atomic OpenShift Master API
Documentation=https://github.com/openshift/origin
After=etcd_container.service
Wants=etcd_container.service
Before=origin-node.service
After=docker.service
After=chronyd.service
After=ntpd.service
PartOf=docker.service
Requires=docker.service

[Service]
EnvironmentFile=/etc/sysconfig/origin-master-api
Environment=GOTRACEBACK=crash
ExecStartPre=-/usr/bin/docker rm -f origin-master-api
ExecStart=/usr/bin/docker run --rm --privileged --net=host \
  --name origin-master-api \
  --env-file=/etc/sysconfig/origin-master-api \
  -v /var/lib/origin:/var/lib/origin \
  -v /var/log:/var/log -v /var/run/docker.sock:/var/run/docker.sock \
  -v /etc/origin:/etc/origin \
  \
  -v /etc/pki:/etc/pki:ro \
  \
  openshift/origin:${IMAGE_VERSION} start master api \
  --config=${CONFIG_FILE} $OPTIONS
ExecStartPost=/usr/bin/sleep 10
ExecStop=/usr/bin/docker stop origin-master-api
LimitNOFILE=131072
LimitCORE=infinity
WorkingDirectory=/var/lib/origin
SyslogIdentifier=origin-master-api
Restart=always
RestartSec=5s

[Install]
WantedBy=docker.service
WantedBy=origin-node.service
```

**Comment by ceagan** [comment]
I'm not sure if this in relevant or not, but I did have to change the hostnames of these machines to get past an earlier error. The hostnames were fully-qualified in the inventory file, but the short hostname was returned when queried with `uname -n`. I ran into a problem where the node configurations being moved to config maps caused the node name to not be set anymore in the node config files. This caused the nodes to fail to register themselves properly when joining the cluster. I changed the hostnames of all the systems in the cluster to use the fully-qualified names with `hostnamectl set-hostname` and now `uname -n` returns the fully-qualified names. That resolved the node config issue, but I suppose there could still be something in the Ansible scripts that expects the hostname call to return the short name...

**Comment by ceagan** [comment]
Should _roles/openshift_control_plane/tasks/upgrade.yml_ include _- import_tasks: static.yml_ someplace?

**Comment by ceagan** [comment]
After more research into this problem, I've decided that these things are not working quite right.

1. The origin-master-api and origin-master-controllers service files for v3.9 were not deleted, even though the Ansible scripts appear to want them to be deleted. Event though they still exist, the origin-master-api environment file they need was deleted, so they fail to start.
2. The migrated environment file based on the origin-master-api sysconfig still refers to version 3.9.
3. The /etc/origin/node/pods directory only has an etcd.yaml file and I believe it is supposed to have an apiserver.yaml and a controller.yaml file.

I'm still thinking that some of the tasks in the _roles/openshift_control_plane/tasks/static.yaml_ file needs to be run.

Can anyone on the project confirm these suspicions?

I'm simply trying to upgrade a v3.9 installation to v3.10 so I'm surprised I'm the only one running into this problem. :(

**Comment by ceagan** [comment]
I'm probably the only one who ran into this specific problem, but I'll leave this solution here for anyone who might find it useful.

I believe that the root cause was due to my earlier failure with the hostnames. That failure happened at a point which left the first master half-upgraded with the Master API not running. The _static.yaml_ file was probably not run in my case, leaving a Master API that didn't work when run in either the old or new methods. Since the Ansible scripts that upgrade the master nodes expect a working Master API before they will continue, I reverted the configuration back to use v3.9 by replacing the _/etc/origin/master/master-config.yaml_ with a backup and restoring the missing environment file _/etc/sysconfig/origin-master-api_.

I then ensured that the v3.9 Master API service was up and running. This allowed the scripts to continue.
